test/e2e/push_test.go

```diff
@@ -63,4 +63,127 @@ var _ = Describe("Podman push", func() {
+		debug.WaitWithDefaultTimeout()
+
+		session = podmanTest.Podman([]string{"run", "-d", "-p", "5000:5000", "--name", "registry", "-v",
+			strings.Join([]string{authPath, "/auth"}, ":"), "-e", "REGISTRY_AUTH=htpasswd", "-e",
+			"REGISTRY_AUTH_HTPASSWD_REALM=Registry Realm", "-e", "REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd",
+			"-v", strings.Join([]string{certPath, "/certs"}, ":"), "-e", "REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt",
+			"-e", "REGISTRY_HTTP_TLS_KEY=/certs/domain.key", "registry:2"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		session = podmanTest.Podman([]string{"logs", "registry"})
+
+		login := podmanTest.Podman([]string{"login", "-u", "podmantest", "-p", "test", "localhost:5000"})
+		login.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		push := podmanTest.Podman([]string{"push", ALPINE, "localhost:5000/my-alpine"})
+		push.WaitWithDefaultTimeout()
+		Expect(push.ExitCode()).To(Equal(0))
+	})
+
+	It("podman push to docker-archive", func() {
+		session := podmanTest.Podman([]string{"push", ALPINE, "docker-archive:/tmp/alp:latest"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		clean := podmanTest.SystemExec("rm", []string{"/tmp/alp"})
+		clean.WaitWithDefaultTimeout()
+		Expect(clean.ExitCode()).To(Equal(0))
+	})
+
+	It("podman push to docker daemon", func() {
+		setup := podmanTest.SystemExec("bash", []string{"-c", "systemctl status docker 2>&1"})
+		setup.WaitWithDefaultTimeout()
+
+		if setup.LineInOuputContains("Active: inactive") {
+			setup = podmanTest.SystemExec("systemctl", []string{"start", "docker"})
+			setup.WaitWithDefaultTimeout()
+
+			defer podmanTest.SystemExec("systemctl", []string{"stop", "docker"})
+		} else if setup.ExitCode() != 0 {
+			Skip("Docker is not avaiable")
+		}
+
+		session := podmanTest.Podman([]string{"push", ALPINE, "docker-daemon:alpine:podmantest"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+
+		check := podmanTest.SystemExec("docker", []string{"images", "--format", "{{.Repository}}:{{.Tag}}"})
+		check.WaitWithDefaultTimeout()
+		Expect(check.ExitCode()).To(Equal(0))
+		Expect(check.OutputToString()).To(ContainSubstring("alpine:podmantest"))
+
+		clean := podmanTest.SystemExec("docker", []string{"rmi", "alpine:podmantest"})
+		clean.WaitWithDefaultTimeout()
+		Expect(clean.ExitCode()).To(Equal(0))
+	})
+
+	It("podman push to oci-archive", func() {
+		session := podmanTest.Podman([]string{"push", ALPINE, "oci-archive:/tmp/alp.tar:latest"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
+		clean := podmanTest.SystemExec("rm", []string{"/tmp/alp.tar"})
+		clean.WaitWithDefaultTimeout()
+		Expect(clean.ExitCode()).To(Equal(0))
+	})
+
+	It("podman push to local ostree", func() {
+		setup := podmanTest.SystemExec("which", []string{"ostree"})
```

Comment:
This one will probably need to be gated on OSTREE support.

Reply:
Do we have a build tag for this?

Reply:
cat hack/ostree_tag.sh 
#!/bin/bash
if ! pkg-config ostree-1 2> /dev/null ; then
	echo containers_image_ostree_stub
fi

We could create a ostree test, and use this buildtag.

Reply:
OK. Thanks for your comments @rhatdan . Will update this code.

Reply:
Hi @rhatdan 
Go through the code and didn't find any variable about "OSTREE", can you show me more information about it? Thanks a lot.

Reply:
Maybe you don't have to worry about it.